src/model.cpp

```diff
@@ -588,6 +880,7 @@ void Model::fsigma_z(const realtype t, const int ie, const int *nroots,
 void Model::fdsigma_zdp(const realtype t) {
     std::fill(dsigmazdp.begin(),dsigmazdp.end(),0.0);
     for(int ip = 0; (unsigned)ip < plist_.size(); ip++)
+        // TODO dsigmazdp.data() or slice?
```

Comment:
fdsigma_zdp code should be roughly the equivalent code as for fdsigma_ydp (that might currently not have been the case), the only thing that changes is that for individual ne, we have to check z2event for which iz we may write values.